contracts/tests/sbtc-token.test.ts

```diff
@@ -399,17 +400,159 @@ describe("sBTC token contract", () => {
       expect(receipt.value).toEqual(errors.registry.ERR_UNAUTHORIZED);
     });
 
+    test("Fail a non-protocol principal calling protocol-locked", () => {
+      const receipt = txErr(
+        token.protocolLock({
+          amount: 1000n,
+          owner: bob,
+        }),
+        bob
+      );
+      expect(receipt.value).toEqual(errors.registry.ERR_UNAUTHORIZED);
+    });
+
+    test("Fail a non-protocol principal calling protocol-burn", () => {
+      const receipt = txErr(
+        token.protocolBurn({
+          amount: 1000n,
+          owner: bob,
+        }),
+        bob
+      );
+      expect(receipt.value).toEqual(errors.registry.ERR_UNAUTHORIZED);
+    });
+
+    test("Fail a non-protocol principal calling protocol-burn-locked", () => {
+      const receipt = txErr(
+        token.protocolBurnLocked({
+          amount: 1000n,
+          owner: bob,
+        }),
+        bob
+      );
+      expect(receipt.value).toEqual(errors.registry.ERR_UNAUTHORIZED);
+    });
+
     test("Fail transferring sbtc when not owner", () => {
       const receipt1 = txErr(
         token.transfer({
           amount: 999n,
           sender: alice,
           recipient: bob,
-          memo: new Uint8Array(1).fill(0),
+          memo: null,
         }),
         bob
       );
       expect(receipt1.value).toEqual(errors.token.ERR_NOT_OWNER);
     });
   });
+
+  describe("protocol actions", () => {
```

Comment:
This only works in non-mainnet

Reply:
Not sure I follow? Does this mean this logic won't hold in mainnet/is broken or that this test only works in non-mainnet tests?

Reply:
This test only works in testnet because the deployer has protocol permissions

Reply:
Ah, right, I follow.